src/ORM/DataObject.php

```diff
@@ -4186,6 +4187,32 @@ public function mergeRelatedObjects($list, $items)
         return $added;
     }
 
+    /**
+     * Generate a unique key for data object
+     * the unique key uses the @see DataObject::getUniqueKeyComponents() extension point so unique key modifiers
+     * such as versioned or fluent are covered
+     * i.e. same data object in different stages or different locales will produce different unique key
+     *
+     * recommended use:
+     * - when you need unique key for caching purposes
+     * - when you need unique id on the front end (for example JavaScript needs to target specific element)
+     *
+     * @return string
+     * @throws Exception
+     */
+    public function getUniqueKey(): string
+    {
+        /** @var UniqueKeyInterface $service */
+        $service = UniqueKeyService::singleton();
+        if (!$service instanceof UniqueKeyInterface) {
```

Comment:
I think this would need to be `!$service instanceof UniqueKeyInterface`

Regardless though, I think we should just remove this block, seems like there's no valid reason why a UniqueKeyService should not be an instance of a UniqueKeyInterface, I think an unhandled exception would be fine in that case

Reply:
It works this way, but I agree. You can do `Injector::inst()->get(UniqueKeyInterface::class)` and define the implementation of that interface in YAML config as UniqueKeyService. We do that with the LoggerInterface for example.

Reply:
It's really the `return bin2hex(random_bytes(16));` that I don't think we need.  Again, if you haven't defined your UniqueKeyInterface properly then I think you should get hard exception, rather than it failing gracefully which it seems like the intention is here.